Packs/Fortanix-DSM/Integrations/FortanixDSM/FortanixDSM.py

```diff
@@ -0,0 +1,579 @@
+    else:
+        cipher = payload_cipher
+
+    if not iv and 'iv' in demisto.args():
+        iv = demisto.args()['iv']
+
+    if not key_id:
+        key_name = PROTECTION_KEY
+        key_id = None
+        if 'key' in demisto.args():
+            key_id = demisto.args()['key']
+
+    if not mode:
+        mode = PROTECTION_MODE
+        if 'mode' in demisto.args():
+            mode = demisto.args()['mode']
+
+    path = 'crypto/v1/decrypt'
+    body = {
+        'cipher': cipher,
+        'alg': 'AES',
+        'mode': mode,
+        'key': {}
+    }
+    if key_name:
+        body['key']['name'] = key_name
+    else:
+        body['key']['kid'] = key_id
+    if iv:
+        body['iv'] = iv
+
+    res = send_request(path, "post", body)
+    value = ''
+    try:
+        if 'plain' in res:
+            if isBase64(res['plain']):
+                value = b64decode(res['plain']).decode("utf-8")
+            else:
+                value = res['plain']
+        else:
+            value = res
+    except Exception:
+        value = res['plain']
+
+    mapped_value = [{
+        'Plain': value
+    }]
+    readable_output = tableToMarkdown(f'Fortanix DSM Decryption {mapped_value}', mapped_value)
+    return CommandResults(
+        outputs_prefix='Fortanix.Data.Plain',
+        readable_output=readable_output,
+        outputs=mapped_value
+    )
+
+
+''' GLOBAL VARIABLES '''
+
+
+# Disable insecure warnings
+urllib3.disable_warnings()  # pylint: disable=no-member
+
+# built-in method to procure proxy from ENV
+handle_proxy()
+
+
+USERNAME = ''
+PASSWORD = ''
+CREDENTIALS = demisto.params().get('credentials', {})
+if CREDENTIALS:
+    USERNAME = CREDENTIALS.get('identifier', '')
+    PASSWORD = CREDENTIALS.get('password', '')
+
+BEARER = ''
+TOKEN = demisto.params().get('token', '')
+VERIFY_SSL = not demisto.params().get('tls_enforce', False)
+
+BASE_URL = get_server_url()
+SERVER_URL = BASE_URL  # + '/v1'
+
+DEFAULT_STATUS_CODES = {999}
+
+GROUP_IDS = demisto.params().get('group_ids', '')
+if GROUP_IDS and GROUP_IDS != '':
+    GROUP_IDS = GROUP_IDS.strip()
+
+PROTECTION_KEY = demisto.params().get('protection_key', None)
+PROTECTION_MODE = demisto.params().get('protection_mode', None)
+
+
+''' MAIN FUNCTION '''
+
+
+def main() -> None:
+
+    demisto.debug('Executing command: ' + demisto.command())
+    message = ''
+
+    try:
+        if (USERNAME and PASSWORD and USERNAME != '' and PASSWORD != '') or (TOKEN and TOKEN != ''):
+            BEARER = login()
+        elif BEARER == '' or TOKEN == '':
+            return_results('Either an API KEY or App/User credentials must be provided')
+
+        if demisto.command() == 'test-module':
+            test_command(demisto.command())
+        elif demisto.command() == 'fortanix-list-secrets':
+            return_results(list_secrets_command())
+        elif demisto.command() == 'fortanix-get-secret-metadata':
+            return_results(list_secrets_command())
```

Comment:
yes, the function checks if the args contains a kid to lookup or if the result set contains a single key and in both cases returns the key metadata. Otherwise it sends a list (table) of keys that are found.